### PR TITLE
Add frontend media documentation

### LINKED ISSUE #4025: Write TSDoc to document frontend fields
## Description

For fields with complications, or edge cases, or noteworthy details, write descriptions in the frontend interfaces that can be parsed by the documentation generation script from #4023.

Also document what the frontend expects from the API, what it wrangles and generates automatically.

## Instructions

A quick primer about the media documentation pipeline for the frontend: the `just` recipe `frontend/generate-docs` calls `pnpm` script `doc:media-props` which runs `frontend/scripts/document-media.js`. This JS script parses `frontend/types/media.ts` and writes a Markdown file that's served by the docs.

So to do this, you'll need to follow these steps.

- Start the docs server `just s`, let it run in the background.
- Edit the sole file `frontend/types/media.ts`.
  - Add documentation comments `/** */` for properties of the interfaces `Media`, `ImageDetail` and `AudioDetail`.
  - You can write Markdown, you can use TSDoc features like `@see`.
  - You can also see refer to pre-written docstrings (e.g. `id` and `title`) as examples.
- Run `just frontend/generate-docs media-props false`.
- Check the running docs @ `/meta/media_properties/frontend.html` to see your updates on the page.

### PR DESCRIPTION
**Pull request by zackkrida**
## Fixes

Fixes #4025 by @dhruvkb 

## Description

Adds new comments to media properties in documentation/meta/media_properties/frontend.md, and publishes the generated documentation changes.

I tried to provide information relevant to the context of using these properties in the frontend, rather than purely generic information. I am very open to suggestions on what is useful and consider this a PR that will generate a lot of positive feedback as a "conversation starter" rather than a perfect implementation.

## Testing Instructions

Review the produced docs.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [x] I ran the DAG documentation generator (`just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`just catalog/generate-docs media-props`
      for the catalog or `just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/src/types/media.ts [hunk 1]
```diff
@@ -94,31 +120,44 @@ export interface AudioSet {
 
 export interface AudioDetail extends Media {
   frontendMediaType: "audio"
-
+  /** An object representing a group of audio tracks this track belongs to, like an album or podcast. */
   audio_set?: AudioSet
+  /** A raw list of strings representing musical genres. */
   genres?: string[]
   length?: string
```
Comment (zackkrida) [inline frontend/src/types/media.ts:127]: I do not actually know what this is :relaxed:

File: frontend/src/types/media.ts [hunk 2]
```diff
@@ -35,50 +36,133 @@ export interface Media {
...
+  /** The full text to properly attribute the work to the creator. */
   attribution: string
 
   frontendMediaType: SupportedMediaType
 
+  /** A long test describing the media, from the provider. */
   description?: string
 
-  category: string | null
+  /** A list of descriptive categories describing the work. */
```
Comment (obulat) [inline frontend/src/types/media.ts:75]:
```suggestion
  /** The descriptive category describing the work. */
```

There's actually an Enum of possible categories in the API.

File: frontend/src/types/media.ts [hunk 3]
```diff
@@ -35,50 +36,133 @@ export interface Media {
...
+  category?: string | null
+
+  /**
+   * A snake_cased slug representing the media's provider.
+   *
+   * Corresponds to the `source_name` field of provider results
+   * from the API's `/stats/` endpoints.
+   */
```
Comment (obulat) [inline frontend/src/types/media.ts:83]: Nice!

File: frontend/src/types/media.ts [hunk 4]
```diff
@@ -35,50 +36,133 @@ export interface Media {
...
+   * If true, the media has sensitivity markers that might require user discretion or warnings.
+   */
   isSensitive: boolean
 }
 
 export interface ImageDetail extends Media {
   frontendMediaType: "image"
-
-  /** the vertical length of the image in pixels */
+  /** the vertical length of the image, in pixels. */
```
Comment (obulat) [inline frontend/src/types/media.ts:163]:
```suggestion
  /** The vertical length of the image, in pixels. */
```

File: frontend/src/types/media.ts (hunk 1 above)
Comment (obulat) [reply frontend/src/types/media.ts:188]: This should be removed.
`length` is a query parameter, not a property of an `AudioDetail`.

https://github.com/WordPress/openverse/blob/292946e588e839e1e305f2bf163986020b738303/frontend/src/utils/search-query-transform.ts#L38

File: frontend/src/types/media.ts [hunk 5]
```diff
@@ -95,30 +179,52 @@ export interface AudioSet {
...
   audio_set?: AudioSet
+
+  /** A raw list of strings representing musical genres. */
   genres?: string[]
+
   length?: string
-  /** the time period of the track in milliseconds */
+
+  /**
+   * The time period of the track in milliseconds.
```
Comment (obulat) [inline frontend/src/types/media.ts:191]:
```suggestion
   * The duration of the track in milliseconds.
```
Duration seems clearer than "time period" here

File: frontend/src/types/media.ts [hunk 6]
```diff
@@ -95,30 +179,52 @@ export interface AudioSet {
...
+   * This provides the duration of the audio track in a numerical format.
+   */
   duration?: number
+
   /**
-   * amount of digital audio data transmitted or processed in unit time; This
-   * field holds numbers measured in bits per second.
+   * Amount of digital audio data transmitted or processed in unit time;
```
Comment (obulat) [inline frontend/src/types/media.ts:197]:
```suggestion
   * Amount of digital audio data transmitted or processed in unit time.
```

File: frontend/src/types/media.ts [hunk 7]
```diff
@@ -95,30 +179,52 @@ export interface AudioSet {
...
+   * This field holds numbers measured in bits per second (bps).
    *
    * @see {@link https://en.wikipedia.org/wiki/Bit_rate#Audio|Wikipedia}
    */
   bit_rate?: number
+
   /**
-   * number of samples for digital representation taken in unit time; This field
-   * holds numbers measured in hertz.
+   * Number of samples for digital representation taken in unit time;
```
Comment (obulat) [inline frontend/src/types/media.ts:205]:
```suggestion
   * Number of samples for digital representation taken in unit time.
```

**Comment by AetherUnbound** [comment]
Removing myself and Staci from reviewing this as we don't have enough subject-matter expertise or context for it.

**Review by obulat — APPROVED** [review]
Good start for this implementation. I added several suggestions, but don't wan't to block.

I feel like it's wasteful to re-write the comments for each layer of the stack instead of re-using the docs from catalog/API. I don't know what the best implementation of this would be.

**Comment by zackkrida** [comment]
Thanks @obulat! I agree with your suggestions and the redundancy of this approach. I'll apply your suggestions. Once we get this merged (which will be an improvement over the current lack of docs) we can consider ways of reducing redundancy like having root TypeScript types that are transpiled to Python or something

**Comment by obulat** [comment]
> (which will be an improvement over the current lack of docs)

Agree, that's why I approved here :)

File: frontend/src/types/media.ts [hunk 8]
```diff
@@ -95,30 +179,52 @@ export interface AudioSet {
...
+
+  /** A raw list of strings representing musical genres. */
   genres?: string[]
+
   length?: string
-  /** the time period of the track in milliseconds */
+
+  /**
+   * The time period of the track in milliseconds.
+   * This provides the duration of the audio track in a numerical format.
```
Comment (zackkrida) [inline frontend/src/types/media.ts:192]: ``
```suggestion
```

**Comment by zackkrida** [comment]
@dhruvkb could I get a review here this week?

File: frontend/src/types/media.ts [hunk 9]
```diff
@@ -35,50 +36,133 @@ export interface Media {
    * - handle empty titles
    */
   title: string
-  /** the raw name of the creative work, as returned by the API */
+
+  /** The raw name of the creative work, as returned by the API. */
```
Comment (dhruvkb) [inline frontend/src/types/media.ts:40]: Minor nit/personal preference: This isn't a sentence so it shouldn't start with an uppercase letter and should not end with a fullstop. I'm basing this on how dictionaries present definitions.

**Review by dhruvkb — APPROVED** [review]
LGTM! I have a small nit about punctuation but that is leaning a lot towards personal preference so I'll not block.

**Pull request merged**